protocol.md

```diff
@@ -1323,6 +1324,45 @@ interface LogMessageParams {
 
 Where type is defined as above.
 
+#### <a name="window_progress"></a>Progress Notification
+
+The progress notification is sent from the server to the client to ask the client to indicate progress.
+
+_Notification_:
+* method: 'window/progress'
+* params: `ProgressParams` defined as follows:
+
+```typescript
+interface ProgressParams {
+  /**
+   * A unique identifier to associate multiple progress notifications with the same progress.
+   */
+  id: string;
+
+  /**
+   * The title of the progress.
+   * This should be the same for all ProgressParams with the same id.
+   */
+  title: string;
+
+  /**
+   * Optional progress message to display.
+   */
+  message?: string;
+
+  /**
+   * Optional progress percentage to display.
+   */
+  percentage?: number;
+
+  /**
+   * Set to true on the final progress update.
+   * No more progress notifications with the same ID should be sent.
```

Comment:
We only need one way to signal finished.

Even if the client does not show progress, we can use `progress: 100` to indicate `done`.

The `done` parameter is unnecessary.